src/pages/index.js

```diff
@@ -1,75 +1,117 @@
-const profileTitle = document.querySelector(".profile__title");
-const profileDescription = document.querySelector(".profile__description");
-const profileTitleInput = document.querySelector("#profile-title-input");
-const profileDescriptionInput = document.querySelector(
-  "#profile-description-input"
-);
-const profileEditForm = profileEditModal.querySelector(".modal__form");
+import PopUpWithConfirmation from "../components/PopUpWithConfirmation.js";
 
 const cardTemplate =
   document.querySelector("#card-template").content.firstElementChild;
 const cardListEl = document.querySelector(".cards__list");
 
-/* ------------------------------------------------------------------------------- */
-
-/* Add Buttons */
-const addCardButton = document.querySelector(".profile__add-button");
-const addCardForm = document.querySelector(".modal__card-form");
-const addNewCardModal = document.querySelector("#add-card-modal");
-
-/* ------------------------------------------------------------------------------- */
-
-/* Preview Elements */
-const previewImage = document.querySelector(".modal__preview-image");
-const previewDescription = document.querySelector(
-  ".modal__preview-description"
-);
-
 const cardSelector = "#card-template";
 
-/* ------------------------------------------------------------------------------- */
-
 const userInfo = new UserInfo({
   profileTitle: ".profile__title",
   profileDescription: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
-/* ------------------------------------------------------------------------------- */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "b0f5599c-4007-4941-89b3-30100c4f8838",
+    "Content-Type": "application/json",
+  },
+});
 
-/* Functions */
+let section;
 
-addCardButton.addEventListener("click", () => {
-  addModal.open();
-});
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([cards, data]) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (data) => {
+          const cardEl = renderCard(data);
+          section.addItem(cardEl);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
 
-/* ------------------------------------------------------------------------------- */
+    userInfo.setUserInfo({
+      title: data.name,
+      description: data.about,
+    });
+
+    userInfo.setAvatar({ avatar: data.avatar });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+if (profileAddButton) {
+  profileAddButton.addEventListener("click", () => {
+    addModal.open();
+  });
+} else {
+  console.error("profileAddButton not found in the DOM");
+}
 
 /* Event Listener */
 profileEditButton.addEventListener("click", () => {
   profileTitleInput.value = userInfo.getUserInfo().title;
-  profileDescriptionInput.value = userInfo.getUserInfo().title;
+  profileDescriptionInput.value = userInfo.getUserInfo().description;
```

Comment:
No need in calling `userInfo.getUserInfo()` 2 times


```js


  const { description, title } = userInfo.getUserInfo();
  profileTitleInput.value = title;
  profileDescriptionInput.value = description;
```

Reply:
* Another remark hasn't been corrected https://github.com/mowooje/se_project_aroundtheus/pull/4#discussion_r1759991328